### PR TITLE
Enable xdebug for local debugging.

### PR DESCRIPTION
**Pull request by Firesphere**
Before going completely crazy. Here's the final version with only one commit.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -26,6 +26,7 @@ Vagrant.configure("2") do |config|
   config.vm.provision :shell, :path => "scripts/apache.sh"
   config.vm.provision :shell, :path => "scripts/mariadb.sh"
   #config.vm.provision :shell, :path => "scripts/php-mcrypt.sh"
+  config.vm.provision :shell, :path => "scripts/xdebug.sh"
```
Comment (dhensby) [inline Vagrantfile]: Do we want it enabled by default?

File: scripts/xdebug.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+echo "Install Xdebug"
+yum install -y gcc gcc-c++ autoconf automake --enablerepo=epel --enablerepo=webtatic
+
+pecl install Xdebug
+if [ ! -f /etc/php.d/xdebug.ini ]
+then
+    echo "[xdebug]
+    zend_extension=\"/usr/lib64/php/modules/xdebug.so\"
+    xdebug.remote_connect_back = 1
+    xdebug.remote_enable = 1" >> /etc/php.d/xdebug.ini
```
Comment (dhensby) [inline scripts/xdebug.sh]: This can be a single `>`

File: scripts/xdebug.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+# 1. Go to Settings => Language & Frameworks => PHP => Servers
+# 2. Add you Vagrant server with the correct port.
+# 3. Check the pathmapping checkbox and set your root of the project to /vagrant and the www directory to /www/directory
+# 4. Apply and close settings
+# 5. Go to Run => Edit configurations
+# 6. Add a new PHP Web Application with the server you just added. The rest is trivial and free to fill.
+# 7. Aply and done!
+
+echo "Install Xdebug"
+yum install -y gcc gcc-c++ autoconf automake --enablerepo=epel --enablerepo=webtatic
```
Comment (dhensby) [inline scripts/xdebug.sh]: Do we have to use the epel and webtatic repos here?

If so, you need to include the installer script too (to ensure they are installed)

File: scripts/xdebug.sh (hunk 2 above)
Comment (Firesphere) [inline scripts/xdebug.sh]: Matter of habits I guess? Is it a "serious" issue with this PR?

File: Vagrantfile (hunk 1 above)
Comment (Firesphere) [inline Vagrantfile]: Well, I do, but it might be a matter of what you guys prefer ofcourse.

File: scripts/xdebug.sh (hunk 3 above)
Comment (Firesphere) [inline scripts/xdebug.sh]: I don't think they are needed, included to be sure, but I'd have to check.

File: Vagrantfile (hunk 1 above)
Comment (dhensby) [inline Vagrantfile]: I'd opt for commented out as default

File: scripts/xdebug.sh (hunk 2 above)
Comment (dhensby) [inline scripts/xdebug.sh]: If you're going to edit this PR anyway, lets change it to single `>`

File: scripts/xdebug.sh (hunk 3 above)
Comment (dhensby) [inline scripts/xdebug.sh]: Please do, either way you need to remove it or include the webtatic/epel script.

**Comment by willmorgan** [comment]
A rebase is now required - sorry @Firesphere!

File: scripts/xdebug.sh (hunk 3 above)
Comment (Firesphere) [inline scripts/xdebug.sh]: They're indeed not needed. Copy-paste. Saw it in the copy-paste. Figured I'd leave it. I'll get things sorted in the rebase.

File: scripts/xdebug.sh (hunk 2 above)
Comment (Firesphere) [inline scripts/xdebug.sh]: Will do.

File: Vagrantfile (hunk 1 above)
Comment (Firesphere) [inline Vagrantfile]: Ok, will do.

**Comment by Firesphere** [comment]
Heh, no problem. I'll get too it this weekend I hope. A bit busy atm with some bugs for a very valueable customer.

**Comment by Firesphere** [comment]
Rebase and changes in a new commit or squashed? (Please say new commit, my previous rebase/squash experience left me horrified and in shock of how it didn't do what I wanted)

**Comment by Firesphere** [comment]
I couldn't rebase without new commits. So new commits are now included. I feel stupid :P

**Comment by Firesphere** [comment]
If someone would guide me to a way to squashing this, I'd be so happy. I feel like I only make it worse :(

File: scripts/xdebug.sh [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+# 6. Add a new PHP Web Application with the server you just added. The rest is trivial and free to fill.
+# 7. Aply and done!
+
+echo "Install Xdebug"
+<<<<<<< HEAD
+yum install -y gcc gcc-c++ autoconf automake
+=======
+yum install -y gcc gcc-c++ autoconf automake --enablerepo=epel --enablerepo=webtatic
+>>>>>>> bd45802707205f028b37227aa70fa1eac89494a4
+
```
Comment (Firesphere) [inline scripts/xdebug.sh]: Why, FFS, did this not show up in my conflicts?!

**Comment by dhensby** [comment]
Wuh oh!

@Firesphere ok, so you need to get your code back to before it went all crazy.

The best way to do that is to do `git reflog` and find the hash you had before you started messing around with merges / rebases (if you can't find it, just paste me your output from reflog). In fact, I think it might be this one: https://github.com/Firesphere/vagrant-skeleton/commit/bd45802707205f028b37227aa70fa1eac89494a4

When you find the commit hash to go back to do:

```bash
$ git reset --hard bd45802707205f028b37227aa70fa1eac89494a4
$ git pull --rebase upstream master
$ # fix conflicts
$ git push -f
```

That should be it.

**Comment by dhensby** [comment]
I think (once this is in) that we can tag rc3 and I can't see much need to change for stable...

Though there is that issue with grunt-watch... #69

**Comment by Firesphere** [comment]
I know. SourceTree went totally crazy and made batshit crazy commits and merges.

**Comment by Firesphere** [comment]
Well, I appearantly have some git issues, but this should be it.

**Comment by dhensby** [comment]
Lovely

**Pull request merged**